### PR TITLE
Use algorithm documentation for tooltips on the table

### PR DESCRIPTION
**Pull request by gemmaguest**
This PR improves the tooltips on the per-theta settings table by setting a separate tooltip for each column. Previously, all columns had the same tooltip. The tooltips are taken from the algorithm documentation for the relevant property.

Note that the tooltips are only set on the table cells. The table headers all have the original default tooltip, which explains the context of what the table does.

This is a useful change in itself but is also preparative work for issue #27826 which will add further complexity to the per-theta defaults table, making the tooltips more important.

Refs #27826

**To test:**

- Open the `ISIS Reflectometry` interface and set the instrument to `INTER`
- On the Experiment tab, check that the tooltips for the table display column-specific information for the table cells but a generic description of the table for the column headers.
- Try adding new rows with the Add Row button or deleting rows with the Delete key. Check the tooltips remain correct.
- Enter some information e.g. FirstTransmissionRun = `13463`
- Back on the Runs tab, enter Run = `13460` and angle = `0.5` and click Process. Check that the reduction completes without error.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/scientific_interfaces/ISISReflectometry/Reduction/PerThetaDefaults.h [hunk 1]
```diff
@@ -40,8 +43,25 @@ class MANTIDQT_ISISREFLECTOMETRY_DLL PerThetaDefaults {
...
 
-  static auto constexpr OPTIONS_TABLE_COLUMN_COUNT = 9;
-  using ValueArray = std::array<std::string, OPTIONS_TABLE_COLUMN_COUNT>;
+  // The property name associated with each column. Unfortunately in
+  // QtBatchView we are still using ReflectometryReductionOneAuto to get the
+  // tooltips rather than ReflectometryISISLoadAndProcess. If we change it we
+  // will break the Encoder and Decoder unit tests because C++ tests cannot use
+  // python algorithms. The code needs to be reorganised to avoid creating the
+  // algorithm, or the tests rewritten in python. This only affects tooltips
+  // for a couple of the properties so is not an urgent fix, so for now just
```
Comment (giovannidisiena) [inline qt/scientific_interfaces/ISISReflectometry/Reduction/PerThetaDefaults.h]: Is there an issue open for this?

**Review by giovannidisiena — COMMENTED** [review]
Looks good, just a small comment and waiting on a build for manual testing.

**Comment by giovannidisiena** [comment]
@gemmaguest just to confirm, the column headers should all display the same tooltip re the table as a whole?

File: qt/scientific_interfaces/ISISReflectometry/Reduction/PerThetaDefaults.h (hunk 1 above)
Comment (gemmaguest) [reply qt/scientific_interfaces/ISISReflectometry/Reduction/PerThetaDefaults.h]: There is now :) #27889

**Comment by gemmaguest** [comment]
> @gemmaguest just to confirm, the column headers should all display the same tooltip re the table as a whole?

Yes, that's right

**Review by giovannidisiena — APPROVED** [review]
Looks good, detailed comments and works well :shipit:

**Pull request merged**